backend/backend.go

```diff
@@ -313,22 +313,26 @@ func (backend *Backend) createAndAddAccount(
 		if err != nil {
 			return err
 		}
-		if len(configurations) != 1 {
-			// TODO: unified-accounts
-			return errp.New("Watch-only accounts don't support mixed inputs yet")
-		}
-		configuration := configurations[0]
 		accountsConfig := backend.config.AccountsConfig()
 		for _, account := range accountsConfig.Accounts {
-			if account.Configuration.Hash() == configuration.Hash() && account.CoinCode == coin.Code() {
-				return errp.WithStack(ErrAccountAlreadyExists)
+			if account.CoinCode == coin.Code() {
+				// We detect a duplicate account (subaccount in a unified account) if any of the
+				// configurations is already present.
+				for _, config := range account.Configurations {
+					for _, config2 := range configurations {
+						if config.Hash() == config2.Hash() {
```

Comment:
It still doesn't feel right that the Hash method is relying on json.Marshal. very easy to break: it's enough to change a field name in https://github.com/digitalbitbox/bitbox-wallet-app/blob/bdb9c97/backend/signing/configuration.go#L184-L190 and the whole thing will result in a different hash.

Consider reimplementing Hash without using json.